malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java

```diff
@@ -215,13 +247,39 @@ public void onClick(View view) {
                 if(prevAlertTime!=-1){
                     time.setText(prevAlertTime+"");
                 }
+
+                time.setOnKeyListener(new View.OnKeyListener() {
+
+                    @Override
+                    public boolean onKey(View v, int keyCode, KeyEvent event) {
+
+                        try {
+                            if (!time.getText().toString().equals("") &&
+                                    Integer.parseInt(time.getText().toString().trim()) > 100) {
+                                time.setError("Limit Exceeded");
+                                isLimitExceede = true;
+                            }
+                            else {
+                                isLimitExceede = false;
+                            }
+                        }
+                        catch (NumberFormatException exception) {
+                            time.setError("Limit Exceeded");
```

Comment:
Put errors in strings.xml as it is used multiple times. Do for all errors denoted.